src/components/SearchAndFilter/SearchAndFilter.js

```diff
@@ -49,6 +47,39 @@ const SearchAndFilter = ({ filterPanelData, returnSearchData }) => {
     };
   }, [searchContainerActive]);
 
+  // This useEffect sets up listeners so the panel will close if user clicks
+  // anywhere else on the page or hits the escape key
+  useEffect(() => {
+    const closePanel = () => {
+      setFilterPanelHidden(true);
+    };
+
+    const mouseDown = (e) => {
+      // Check if click is outside of filter panel
+      if (!searchAndFilterRef?.current?.contains(e.target)) {
+        // If so, close the panel
+        closePanel();
+      }
+    };
+
+    const keyDown = (e) => {
+      if (e.code === "Escape") {
+        // Close panel if Esc keydown detected
+        closePanel();
+      }
+    };
+
+    // Add listener on document to capture click events
+    document.addEventListener("mousedown", mouseDown);
+    // Add listener on document to capture keydown events
+    document.addEventListener("keydown", keyDown);
+    // return function to be called when unmounted
+    return () => {
+      document.removeEventListener("mousedown", mouseDown);
+      document.removeEventListener("keydown", keyDown);
+    };
+  }, [searchTerm]);
```

Comment:
Out of curiosity, why it has to be recreated when `searchTerm` changes? Effect doesn't seem to use searchTerm.

Reply:
This looks like a bug, have now removed that dep.